mesa/agent.py

```diff
@@ -348,29 +348,103 @@ def agg(self, attribute: str, func: Callable) -> Any:
+        is_single_attr = isinstance(attr_names, str)
+
+        # Branch earlier based on the `handle_missing` option to avoid repeated checks
+        match handle_missing:
+            case "error":
+                if is_single_attr:
+                    return [
+                        self._get_or_raise(agent, attr_names) for agent in self._agents
+                    ]
+                else:
+                    return [
+                        [self._get_or_raise(agent, attr) for attr in attr_names]
+                        for agent in self._agents
+                    ]
+
+            case "default":
+                if is_single_attr:
+                    return [
+                        self._get_with_default(agent, attr_names, default_value)
+                        for agent in self._agents
+                    ]
+                else:
+                    return [
+                        [
+                            self._get_with_default(agent, attr, default_value)
+                            for attr in attr_names
+                        ]
+                        for agent in self._agents
+                    ]
+
+            case "skip":
+                if is_single_attr:
+                    return [
+                        val
+                        for agent in self._agents
+                        if (val := self._get_skip(agent, attr_names)) is not None
+                    ]
+                else:
+                    return [
+                        [
+                            val
+                            for attr in attr_names
+                            if (val := self._get_skip(agent, attr)) is not None
+                        ]
+                        for agent in self._agents
+                    ]
 
-        """
+            case _:
+                raise ValueError(
+                    f"Unknown handle_missing option: {handle_missing}, "
+                    "should be one of 'error', 'skip', or 'default'"
+                )
 
-        if isinstance(attr_names, str):
-            return [getattr(agent, attr_names) for agent in self._agents]
-        else:
-            return [
-                [getattr(agent, attr_name) for attr_name in attr_names]
-                for agent in self._agents
-            ]
+    # Helper method to get an attribute or raise an error if it's missing
+    def _get_or_raise(self, agent: Any, attr_name: str) -> Any:
+        return getattr(agent, attr_name)
+
+    # Helper method to get an attribute with a default value if missing
+    def _get_with_default(self, agent: Any, attr_name: str, default: Any) -> Any:
+        return getattr(agent, attr_name, default)
+
+    # Helper method to skip missing attributes by returning None
+    def _get_skip(self, agent: Any, attr_name: str) -> Any:
+        try:
+            return getattr(agent, attr_name)
+        except AttributeError:
+            return None
```

Comment:
@Corvince you're right. It should return nothing (and add nothing to the list), not `None`.

Sorry, my code, my bad.